app/src/main/java/ingsw/group1/findmyphone/event/SMSLogEvent.java

```diff
@@ -0,0 +1,203 @@
+     *
+     * @see ingsw.group1.findmyphone.pending.PendingSMSEventListener for more details.
+     */
+    public static final String RESULT_KEY = "sms-log-event-result";
+    /**
+     * Message for the error thrown on bad extra.
+     */
+    private static final String EXTRA_ERROR =
+            "The provided String extra was not suitable for the provided type";
+    /**
+     * Message for the error thrown on negative time.
+     */
+    private static final String TIME_ERROR =
+            "The provided time was lower than zero, must be >= 0.";
+
+    /**
+     * Type of event.
+     */
+    @NonNull
+    private EventType eventType;
+
+    /**
+     * Address related to this event, it is relevant to note that this is expected to be a valid
+     * phone number, and is treated as such.
+     */
+    @NonNull
+    private String contactAddress;
+
+    /**
+     * Time of this event.
+     */
+    @NonNull
+    private Long startTime;
+
+    /**
+     * Extras of this event. Should be:
+     * - {@code null} if the Event represents a failed operation, suitable for any event type.
+     * - A valid {@link String} for
+     * {@link ingsw.group1.findmyphone.location.GeoPosition#GeoPosition(String)} if
+     * {@link SMSLogEvent#eventType} is either {@link EventType#LOCATION_REQUEST_SENT} or
+     * {@link EventType#LOCATION_REQUEST_RECEIVED}.
+     * - A valid {@link String} for {@link Long#parseLong(String)} if
+     * {@link SMSLogEvent#eventType} is either {@link EventType#RING_REQUEST_SENT} or
+     * {@link EventType#RING_REQUEST_RECEIVED}, such {@code Long} should be a positive number.
+     */
+    @Nullable
+    private String extra;
+
+    /**
+     * No parameter constructor. Creates an empty event with unknown type, needed for
+     * deserialization from disk.
+     */
+    public SMSLogEvent() {
+        this.eventType = EventType.UNKNOWN;
+        this.contactAddress = "";
+        this.startTime = 0L;
+        this.extra = null;
+    }
+
+    /**
+     * Default constructor.
+     *
+     * @param eventType      The type of event for this instance.
+     * @param contactAddress The address for this event, must be a valid phone number, and is
+     *                       always treated as that, no checks are performed inside this class.
+     * @param startTime      The time at which the event started. Must be higher than or equal to 0.
+     * @param extra          The extra info about this event.
+     * @throws IllegalArgumentException If the extra info returns false for
+     *                                  {@link SMSLogEvent#isValidExtra(EventType, String)} or if
+     *                                  {@code startTime} was lower than 0.
+     */
+    public SMSLogEvent(
+            @NonNull EventType eventType,
+            @NonNull String contactAddress,
+            @NonNull Long startTime,
+            @Nullable String extra) {
+        this.eventType = eventType;
+        this.contactAddress = contactAddress;
```

Comment:
Here i think to avoid generating smslogevent with wrong contactAddress you could try to build a SMSPeer using this address and make it launch an exception if the address is not valid?